### DISCUSSION TITLE
ContextInfo Fields : Mention, Quote, Disappearing Message (Expiration)

### DISCUSSION THREAD

**Discussion by mamur-rezeki** [tulir/whatsmeow#259] [WhatsApp protocol Q&A]
## Mentioning Protocol.

```go
// Example ContextInfo for ExtendedTextMessage

var msg = &waProto.Message{
    ExtendedTextMessage: &waProto.ExtendedTextMessage{
      Text:        proto.String("Metioning"),
      ContextInfo: &waProto.ContextInfo{
        MentionedJid: []string{
          event.Info.Sender.ToNonAD().String(),
        },
      },
    },
  }


```

## Quote

```go

var msg = &waProto.Message{
    ExtendedTextMessage: &waProto.ExtendedTextMessage{
      Text:        proto.String("Quote"),
      ContextInfo: &waProto.ContextInfo{
        StanzaId: proto.String(event.Info.ID),
        Participant: proto.String(event.Info.Sender.ToNonAD().String()),
        QuotedMessage: event.Message,
      },
    },
  }

```

## Disappearing Message (Expiration)

```go

var msg = &waProto.Message{
    ExtendedTextMessage: &waProto.ExtendedTextMessage{
      Text:        proto.String("Disappearing"),
      ContextInfo: &waProto.ContextInfo{
      Expiration: proto.Uint32(86400),
      },
    },
  }

```

**Comment by hoehermann** [comment]
For me, the disappearing message example does not work as expected. Am I using it right like I do [here](https://github.com/hoehermann/purple-gowhatsapp/blob/f9a02220fb99dd89e9d2965c0b87171840637128/src/go/send_message.go#L56)? 

The message is sent and received. However, on the sender the official WhatsApp client says "the receiver updated the message timer". On the receiver it says "you updated the message timer". Somewhere there is a confusion on who set the message timer. 

Also, messages do not disappear – neither on the sender device nor on the receiver.  
**Correction:** The messages actually do disappear. I expected them to disappear after 24 hours sharp. That is not the case, apparently. However, after 26 hours the messages are indeed no longer being displayed. Due to this inaccuracy, smaller values such as "15 Minutes" do not seem to work.

**Reply by mamur-rezeki** [reply]
Maybe if the initial message is sent by us, then we can determine the time rule for the `Expiration` by ourselves.

However, if the initial message is sent by the client, we can take the value of `Expiration` field to be used as the standard expiration time for sending the next message.

**Comment by defryiquartz** [comment]
hey @mamur-rezeki i'm trying to mention the peoples who are mentioned in receiving texts, this is my code. Why is it not working? it's only outputing the text without the mention. Thank you

	mentionedJids := evt.Message.ExtendedTextMessage.ContextInfo.MentionedJid

	var msg = &waProto.Message{
		ExtendedTextMessage: &waProto.ExtendedTextMessage{
			Text: proto.String("Metioning"),
			ContextInfo: &waProto.ContextInfo{
				MentionedJid: mentionedJids,
			},
		},
	}
	_, err := client.SendMessage(ctx, evt.Info.Chat, msg)
	if err != nil {
		return err
	}

	return nil

**Reply by defryiquartz** [reply]
update:
seems like we need to include the @phone into the text as well
example:

	msg := &waProto.Message{
		ExtendedTextMessage: &waProto.ExtendedTextMessage{
			Text: proto.String("Metioning @628123456789"),
			ContextInfo: &waProto.ContextInfo{
				MentionedJid: []string{
					"628123456789@s.whatsapp.net",
				},
			},
		},
	}
	_, err := client.SendMessage(ctx, evt.Info.Chat, msg)
	if err != nil {
		return err
	}

	return nil

**Reply by mamur-rezeki** [reply]
Yes, exactly.

**Comment by GustavoGSA** [comment]
Can I mention everyone in a hidden way? Like  Baileys

**Comment by wibi9424** [comment]
how to make it automatically disappear following group info
